cache.go

```diff
@@ -230,16 +230,23 @@ func (a *Cache[K, V]) SetCapacity(available, max int64) {
 func (a *Cache[K, V]) setCapacity(available, max int64, largerOnly bool) {
 	for {
 		cap := a.cap.Load()
+		size := a.size.Load()
 
-		base := cap
+		base := size
 		if largerOnly {
-			// NOTE: This is legacy, keeping for rollback if needed.
```

Comment:
Have kinda lost track on how the paths diverge now (or don't). Could we make it clearer that SetLargerCapacity is unchanged since before e4e0573d0cee45263514fefa64c9b17a56b4d677?